src/coreclr/vm/mono/mono_coreclr.cpp

```diff
@@ -540,25 +540,16 @@ extern "C" EXPORT_API MonoClassField* EXPORT_CC mono_class_get_field_from_name(M
     }
     CONTRACTL_END;
 
-    while (klass)
+    MonoClass_clr* mt = reinterpret_cast<MonoClass_clr*>(klass);
+    FieldDesc* retVal = NULL;
+    // FindField only checks the fields of the current class and not inherited ones.
+    while (retVal == NULL && mt != NULL)
```

Comment:
had to change this to iterate parent method tables to get tests to pass. FindField only checks current class' fields by design it appears.